src/handlebars/NostoAutocomplete.ts

```diff
@@ -47,27 +50,34 @@ export class NostoAutocomplete extends HTMLElement {
   constructor() {
     super()
   }
+
   async connectedCallback() {
-    const templateElement = this.querySelector<HTMLTemplateElement>("template")
+    return initAutocomplete(this)
+  }
+}
 
-    if (!Object.keys(this.getConfigFromScript()).length) {
-      throw new Error("NostoAutocomplete: Missing required config.")
-    }
+async function initAutocomplete(element: NostoAutocomplete) {
```

Comment:
the initAutocomplete implementations are very similar
move this function to src/lib and parametrize it